### ISSUE TITLE
Improve error message for typo in setter name

### ISSUE THREAD

**Issue by rkj** [dart-lang/sdk#48603]
Code like this:

```dart
class A {}

class B {
  A _a;
  A get a => _a;
  set A(A a) => _a = a;
}
```
gives error:
```
Error: 'A' isn't a type.
  A _a;
```
which is a little confusing, as the type is clearly defined above. It took me a long while to find out what is the actual issue (I first thought incorrect import / conflicting names imported).

It also causes analyzer to highlight all the places except where the actual problem is:
![8pTMspp9ZPUtgWb](https://user-images.githubusercontent.com/4562/158897695-2ad565e2-3514-4b28-9504-0083f8a7c643.png)

**Comment by bwilkerson** [comment]
I agree that the message isn't very helpful in this particular case.

That said, Dart doesn't disallow the use of uppercase names for instance members, so there's nothing wrong with defining a setter named `A`. And, Dart's look-up rules say that `A` should resolve to the instance member rather than a top-level name inside the class that defines the instance member. So technically (though unhelpfully) all of the highlighted usage of the name are places where the analyzer correctly resolved the name to the setter and let you know that a setter wasn't appropriate.

That leaves the question of how to be more helpful. My first attempt to improve the situation would be to include an indication of the definition of the name in the message:

`A` is a setter, but a type was expected here.

We could also provide a context message explaining where the setter `A` is defined.

Given that the common convention is for instance members to start with a lowercase letter, we might also want to change the correction message to include a suggestion that you might want to rename the setter.

Would that be better? Would it be enough, or is there additional information you would have found useful?

**Comment by rkj** [comment]
That would totally help me, specifying what the thing in the error would be probably helpful in other cases.
